### PR TITLE
Stream samples

### PR DESCRIPTION
**Pull request by GeorgeNash**
To avoid out of memory

https://improbableio.atlassian.net/browse/SOL-3371

### CODE REVIEW

File: keanu-project/src/main/java/io/improbable/keanu/util/csv/ColumnWriter.java [hunk 1]
```diff
@@ -35,10 +37,11 @@ public File toFile(File file) {
                 List<Object> flatList = vertex.getValue().asFlatList();
                 row.add(i < flatList.size() ? flatList.get(i).toString() : getEmptyValue());
             }
-            String[] rowToString = new String[row.size()];
-            data.add(row.toArray(rowToString));
+            String[] rowArray = new String[row.size()];
+            writer.writeNext(row.toArray(rowArray), false);
         }
-        return writeToFile(file, data);
+        writer.close();
```
Comment (migwellian) [inline keanu-project/src/main/java/io/improbable/keanu/util/csv/ColumnWriter.java]: `CSVWriter` implements `Closeable` which means you can use "try with resources"

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Comment (migwellian) [reply keanu-project/src/main/java/io/improbable/keanu/util/csv/ColumnWriter.java]: ditto in `RowWriter` and `SampleWriter`

Comment (GeorgeNash) [reply keanu-project/src/main/java/io/improbable/keanu/util/csv/ColumnWriter.java]: done

**Comment by gordoncaleb** [comment]
Have you tried writing huge number of samples with this?

**Comment by GeorgeNash** [comment]
> Have you tried writing huge number of samples with this?

Yeah - dumped 500k samples into a csv

**Pull request merged**